### PR TITLE
improve: effect select option label naming and grouping

### PR DESCRIPTION
**Pull request by doommm**
Re: #

### Changelog

* improve effect select option label naming and grouping
image

### Dependent PR
- https://github.com/cocos/cocos-editor/pull/1713

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: editor/inspector/assets/material.js [hunk 1]
```diff
@@ -4,6 +4,58 @@ const { materialTechniquePolyfill } = require('../utils/material');
...
+    let htmlTemplate = '';
+    let currGroup = '';
+
+    for (const effect of sortedEffects) {
+        const groupName = effectGroupNameRE.exec(effect.path)?.[1] ?? '';
+        let optionLabel = effect.path;
+
+        // complete prev group
+        if (currGroup !== '' && currGroup !== groupName) {
+            htmlTemplate += '</optgroup>\n\n';
```
Comment (arsen2010) [inline editor/inspector/assets/material.js]: 标签不需要 `\n\n`

Comment (arsen2010) [reply editor/inspector/assets/material.js]: 下同

File: editor/inspector/assets/material.js [hunk 2]
```diff
@@ -4,6 +4,58 @@ const { materialTechniquePolyfill } = require('../utils/material');
...
+    for (const effect of sortedEffects) {
+        const groupName = effectGroupNameRE.exec(effect.path)?.[1] ?? '';
+        let optionLabel = effect.path;
+
+        // complete prev group
+        if (currGroup !== '' && currGroup !== groupName) {
+            htmlTemplate += '</optgroup>';
+        }
+
+        if (groupName !== '') {
```
Comment (arsen2010) [inline editor/inspector/assets/material.js:25]: 这个 if 的范围可以包含上面一个 if ；

File: editor/inspector/assets/material.js [hunk 3]
```diff
@@ -4,6 +4,58 @@ const { materialTechniquePolyfill } = require('../utils/material');
...
+            optionLabel = optionLabel.replace(effectGroupNameRE, '');
+        }
+
+        // remove prefix 'effects'(after 'db://' prefix removed)
+        if (effectDirRE.test(optionLabel)) {
+            optionLabel = optionLabel.replace(effectDirRE, '');
+        }
+
+        // use full name as option value
+        htmlTemplate += `<option value="${effect.name}" data-uuid="${effect.uuid}">${optionLabel}</option>\n`;
```
Comment (arsen2010) [inline editor/inspector/assets/material.js]: 末尾不需要 `\n`

**Review by arsen2010 — APPROVED** [review]

**Comment by doommm** [comment]
@VisualSJ

**Review by VisualSJ — APPROVED** [review]

**Pull request merged**